src/realm/object-store/schema.cpp

```diff
@@ -349,20 +366,34 @@ std::vector<SchemaChange> Schema::compare(Schema const& target_schema, SchemaMod
     return changes;
 }
 
-void Schema::copy_keys_from(realm::Schema const& other) noexcept
+void Schema::copy_keys_from(realm::Schema const& other, bool is_schema_additive) noexcept
```

Comment:
This does not seem a hot path, if we prefer to append the objects that did not match within `zip_matching`, I can work out a solution for this. I find processing first the properties and secondly the objects a more clear approach. Also, because `zip_matching` is used in multiple places.

Reply:
This is a hot path. We call this on the fast path of initializing a Realm when we already have another Realm for the file open and can skip most of the initialization work.

Reply:
If this is a hot path, isn't this PR going to slow down the process of understanding what it is needed in order to initialize the second realm instance? 
Since we are basically appending extra fields + extra classes.

Reply:
Yes, currently this looks like it could be a significant performance regression.